spec/std/openssl/ssl/hostname_validation_spec.cr

```diff
@@ -0,0 +1,54 @@
+require "spec"
+require "openssl/ssl/hostname_validation"
+
+describe OpenSSL::SSL::HostnameValidation do
+  describe "matches_hostname?" do
+    it "matches ipv4, ipv6" do
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("192.168.1.1", "192.168.1.1").should be_true
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("192.168.1.2", "192.168.1.1").should be_false
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("::1", "::1").should be_true
+      OpenSSL::SSL::HostnameValidation.matches_hostname?("::1", "::2").should be_false
```

Comment:
I thought you meant the `Socket.ip?` specs.

Matching IPs is out of scope for RFC6125, because it's rare (less than 1%). Here it's a best effort. Yet, I guess I should convert to the "network byte order" representation beforehand...